src/Data/StrMap.purs

```diff
@@ -112,10 +118,20 @@ instance foldableStrMap :: Foldable StrMap where
   foldr f z m = foldr f z (values m)
   foldMap f = foldMap (const f)
 
+instance foldableWithIndex :: FoldableWithIndex String StrMap where
+  foldlWithIndex f = fold (flip f)
+  foldrWithIndex f z m = foldr (uncurry f) z (toArrayWithKey Tuple m)
+  foldMapWithIndex = foldMap
```

Comment:
- `foldlWithIndex`: nearly equivalent to `fold`.  The `flip` converts `k -> acc -> v -> acc` to `k -> v -> acc -> acc`.  (The ordering
- `foldrWithIndex`: first unpack into an array of `Tuple k v`, then use the standard `foldr` for arrays.  The `uncurry` converts `k -> v -> acc -> acc` to `Tuple k v -> acc -> acc` suitable for `foldr`.
- `foldMapWithIndex`: same as `foldMap`.